Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestStepViewController.swift

```diff
@@ -47,4 +54,43 @@ class TestStepViewController: UIViewController {
         indexDelegate.didSetStatus(status: .pass)
         navigationController?.popViewController(animated: true)
     }
+    
+    // Used to set up text window popup, called in viewDidLoad
+    func setupPopup() {
+        NotificationCenter.default.addObserver(self, selector: #selector(TestStepViewController.keyboardWillShow), name: NSNotification.Name.UIKeyboardWillShow, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(TestStepViewController.keyboardWillHide), name: NSNotification.Name.UIKeyboardWillHide, object: nil)
+        inputResultsBox.isHidden = true
+        inputResultsBackground.isHidden = true
+        inputResultsTextBox.delegate = self
+    }
+    
+    // Called when 'Done' button in popup is clicked
+    @IBAction func saveText(_ sender: UIButton) {
+        inputResultsBackground.isHidden = true
+        inputResultsBox.isHidden = true
+        if self.stepResult != inputResultsTextBox.text {
+            self.stepResult = inputResultsTextBox.text
+        }
+        inputResultsTextBox.resignFirstResponder()
+    }
+    
+    // Move popup when keyboard appears/hides
+    func keyboardWillShow(notification: NSNotification) {
+        if let keyboardSize = (notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue)?.cgRectValue {
+            self.inputResultsBox.frame.origin.y -= keyboardSize.height/3
+        }
+    }
+    
+    func keyboardWillHide(notification: NSNotification) {
+        if let keyboardSize = (notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue)?.cgRectValue {
+            self.inputResultsBox.frame.origin.y += keyboardSize.height/3
```

Comment:
In this method I think we need to set the step's result. Currently we are saving whatever is put into the input box into a string but that is never making its way back into the actual step model. I think we should add another function into the protocol defined in the TestRunIndexViewController, something like `didSetStepResult(result: String)` where it sets the steps result value. then call that function from in this method.